wordpress-mobile/WordPress-Android
# [Build] Enable Non-Transitive Resources

This PR is a prerequisite for the `Gradle 8.1.1 & AGP 8.0.2 Upgrade for WPAndroid, WCAndroid & Related Libs` project.

Platform Request: `pdnsEh-13V-p2`
Project Thread: `paaHJt-57Z-p2`

-----

This PR enables non-transitive resources (`android.nonTransitiveRClass`) for the project.

FYI: This behavior becomes the default in AGP `8.0` and higher. As such, this becomes a prerequisite for the AGP `8.0.2` upgrade, that is of course, unless `android.nonTransitiveRClass` is explicitly set to `false`.

PS: I recommend reviewing this PR commit-by-commit, and more so, to focus on the `Build` related commits. Avoid spending too much time on reviewing the `Refactor` related commits as those was purely done for codebase alignment purposes<sup>(*)</sup>.

-----

PS: @AjeshRPai I added you as the main reviewer, randomly so, since I just wanted someone from the WordPress team to be aware of and sign-off on that change for WPAndroid. I also added the @wordpress-mobile/apps-infrastructure team, but this in done only for monitoring purposes, as such, I am not expecting any active review from that team. Thus, feel free to merge this PR if you deem so.

-----

(*) The convention I followed in order to align the R related usage in the codebase is the following:

- To start with, `import org.wordpress.android.R` is the main import of choice.
- Then, and for Kotlin files only, such imports have now the below aliases:
    - `LoginR` for `org.wordpress.android.login.R`
    - `StoriesR` for `com.wordpress.stories.R`
    - `EditorR` for `org.wordpress.android.editor.R`
    - `ZendeskR` for `com.zendesk.sdk.R`
    - `ComposeR` for `androidx.compose.ui.R`
    - `CoordinatorLayoutR` for `androidx.coordinatorlayout.R`
    - `MaterialR` for `import com.google.android.material.R`
    - `AndroidR` for `import android.R`

-----

## To test:

1. Verify that all the CI checks are successful.
2. Thoroughly smoke test both the WordPress and Jetpack apps and check if everything is working as expected.

-----

## Regression Notes

1. Potential unintended areas of impact

    - Potential breakage or misbehaviour of screens as these `R` related changes apply to the whole project, including its module related libraries (like `libs:image-editor` and `libs:editor`).

3. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

4. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

PR submission checklist:


-----

UI Changes testing checklist:
